comp/collector/collector/collectorimpl/agent_check_metadata.go

```diff
@@ -0,0 +1,140 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package collectorimpl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/autodiscoveryimpl"
+	hostMetadataUtils "github.com/DataDog/datadog-agent/comp/metadata/host/hostimpl/utils"
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/collector/externalhost"
+	"github.com/DataDog/datadog-agent/pkg/collector/runner/expvars"
+	"github.com/DataDog/datadog-agent/pkg/serializer/marshaler"
+	jmxStatus "github.com/DataDog/datadog-agent/pkg/status/jmx"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+//
+// The "agent_check" metadata payload contains information about all running checks and the additional hostnames they
+// added to the Agent.
+//
+
+const (
+	defaultInterval   = 10 * time.Minute
+	firstPayloadDelay = 1 * time.Minute
+)
+
+// Payload handles the JSON unmarshalling of the metadata payload
+type Payload struct {
+	hostMetadataUtils.CommonPayload
+	Meta             hostMetadataUtils.Meta `json:"meta"`
+	AgentChecks      []interface{}          `json:"agent_checks"`
+	ExternalhostTags externalhost.Payload   `json:"external_host_tags"`
+}
+
+// MarshalJSON serialization a Payload to JSON
+func (p *Payload) MarshalJSON() ([]byte, error) {
+	// use an alias to avoid infinite recursion while serializing
+	type PayloadAlias Payload
+
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload breaks the payload into times number of pieces
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("AgentChecks Payload splitting is not implemented")
```

Comment:
Do we plan to implement it ? If yes maybe add a "yet" at the end of the sentence ?

Reply:
No, the `SplitPayload` is part of the `serializer` payload interface. It's called when a payload is too big for the intake. But there is no way to "split" a metadata payload into chunk and still be valid.

Reply:
okay thanks